### PR TITLE
feat(input types): Generate update input types with nullable fields

### LINKED ISSUE #17: Generate update input type with no required fields and add validation in resolver

### PR DESCRIPTION
**Pull request by TimMikeladze**
Closes #17

- Improved `addInputTypesForObjectType` function to supporting different input type names and field modifier function. 
- Moved helper functions to `util`
- Updated `ModelDirective` consumption of `addInputTypesForObjectType`

- Added `validateUpdateInputData` to validate null values in an update mutation.
- Updated the `update` mutation in `ModelDirective`.

### CODE REVIEW

File: packages/graphql-crud/src/ModelDirective.ts [hunk 1]
```diff
@@ -68,11 +69,30 @@ export class ModelDirective extends SchemaDirectiveVisitor {
...
-    addInputTypesForObjectType(type, this.schema);
+    addInputTypesForObjectType({
+      objectType,
+      schema: this.schema,
+    });
+
+    // Often times a type will have required fields which the consumer
+    // does not want to include in every update mutation.
+    // In this case we create additional input types for update mutations.
+    // These types are prefixed with `Update`, (for example: `FooUpdateInputType`)
```
Comment (TimMikeladze) [inline packages/graphql-crud/src/ModelDirective.ts]: Fix typo in example

File: packages/graphql-crud/src/validateUpdateInputData.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+// For every null value in the input data
+// check that it can be nullable by check the type definition.
+export const validateUpdateInputData = (props: ValidateUpdateInputDataProps) => {
+  if (isEmpty(props.data)) {
+    throw new Error('data input object is missing');
+  }
+
+  const isNonNullable = (type) => type.astNode.type.kind === 'NonNullType';
```
Comment (alexyuly) [inline packages/graphql-crud/src/validateUpdateInputData.ts]: Why is this function nested? It could be defined at the outermost scope.

Comment (TimMikeladze) [reply packages/graphql-crud/src/validateUpdateInputData.ts]: Agreed. I have it moved to the outermost scope  in an upcoming PR.

**Comment by TimMikeladze** [comment]
Thanks for the review.

**Pull request merged**